src/state/types.ts

```diff
@@ -68,14 +68,22 @@ export type ExtResource = {
   dataSources: Record<string, any>;
 };
 
+export interface UrlOverrides {
+  cluster?: string;
+  namespace?: string;
+  resourceType?: string;
+}
+
+type UrlFunction = (path: string, overrides?: UrlOverrides) => string;
+
 export interface UrlGenerators {
   cluster: string;
   namespace: string;
-  clusterUrl: (path: string) => string;
-  namespaceUrl: (path: string) => string;
-  scopedUrl: (path: string) => string;
-  resourceListUrl: (path: string) => string;
-  resourceUrl: (path: string) => string;
+  clusterUrl: UrlFunction;
+  namespaceUrl: UrlFunction;
+  scopedUrl: UrlFunction;
+  resourceListUrl: UrlFunction;
+  resourceUrl: UrlFunction;
```

Comment:
technically `resourceListUrl` and `resourceUrl` accept the resource as the first parameter (currently just defined as `any` - but should have proper `Resource` type in the future), not string.

Reply:
Technically, we have `K8sResource` type so I will change it to this one